public/js/reports/c3Charts.service.js

```diff
@@ -134,6 +139,17 @@ angular.module('app.reports').service('c3Charts', function (notify, gettextCatal
             tooltip: {
                 order: function (t1, t2) {
                     return t1.id > t2.id;
+                },
+                format: {
+                    value: function (value, ratio, id, index) {
+                        if (dataFormat[id]) {
+                            value = numeral(value).format(dataFormat[id]);
+                        }
+                        if (chart.type === 'donut') {
+                            value = numeral(value).format(report.properties.ykeys[0].format);
+                        }
```

Comment:
There is a possibility that both `if` conditions are true and that `.format()` is called twice.
Check the specific case (donut) first, and do the general case otherwise.

You need to also check if it's a 'pie' report

Reply:
Yes, also I added a if else if and reverse previous condition